algorithms/src/Kokkos_Sort.hpp

```diff
@@ -248,13 +248,13 @@ class BinSort {
 
   BinSort(const_key_view_type keys_, BinSortOp bin_op_,
           bool sort_within_bins_ = false)
-      : BinSort(execution_space{}, keys_, bin_op_, sort_within_bins_) {}
+      : BinSort(exec_space{}, keys_, bin_op_, sort_within_bins_) {}
 
   //----------------------------------------
   // Create the permutation vector, the bin_offset array and the bin_count
   // array. Can be called again if keys changed
-  template <class ExecutionSpace = execution_space>
-  void create_permute_vector(const ExecutionSpace& exec = execution_space{}) {
+  template <class ExecutionSpace = exec_space>
+  void create_permute_vector(const ExecutionSpace& exec = exec_space{}) {
     const size_t len = range_end - range_begin;
```

Comment:
can we make a check here that (i.e. static_assert) that ExecutionSpace can access the memory space of the view?

Reply:
[`c8e2702` (#4919)](https://github.com/kokkos/kokkos/pull/4919/commits/c8e27021125244434b3ef5d4ad2d67ac89bb4cd4)